mirascope/core/base/_utils/_protocols.py

```diff
@@ -113,14 +118,14 @@ def __call__(
         dynamic_config: _BaseDynamicConfigT,
         tools: list[type[BaseTool] | Callable] | None,
         json_mode: bool,
-        call_params: _BaseCallParamsT,
+        call_params:  BaseCallParams[BaseTool],
         extract: bool,
     ) -> tuple[
         CreateFn[_ResponseT, _ResponseChunkT],
         str,
         list[dict[str, Any]],
         list[type[_BaseToolT]] | None,
-        dict[str, Any],
+        BaseCallParams[BaseTool],
```

Comment:
we probably want this to return a generic `_BaseCallKwargsT` that is provider specific (i.e. OpenAI would return `OpenAICallKwargs`, Anthropic would return `AnthropicCallKwargs`, etc...)

Reply:
There's also potentially a solution here where we add a generic to call params so that we can properly type the output of the `base._utils.setup_call` method such that it returns the correct type e.g. `call_params: _BaseCallParamsT[_BaseCallKwargsT]` such that the return can be typed as `_BaseCallKwargsT`.

This way calling `base._utils.setup_call` with an input such as `OpenAICallParams[OpenAICallKwargs]` will properly type the return call_kwargs as `OpenAICallKwargs`

Reply:
I've taken the time to come up with a solution.
I don't think it's feasible with Python's current generic functionality, so I've made it cast after it's been retrun.